deps/libssh2.mk

```diff
@@ -0,0 +1,57 @@
+## libssh2
+
+LIBSSH2_GIT_URL := git://github.com/libssh2/libssh2.git
+LIBSSH2_TAR_URL = https://api.github.com/repos/libssh2/libssh2/tarball/$1
+$(eval $(call git-external,libssh2,LIBSSH2,CMakeLists.txt,build/libssh2.$(SHLIB_EXT),$(SRCDIR)/srccache))
+
+LIBSSH2_OBJ_SOURCE := $(BUILDDIR)/$(LIBSSH2_SRC_DIR)/src/libssh2.$(SHLIB_EXT)
+LIBSSH2_OBJ_TARGET := $(build_shlibdir)/libssh2.$(SHLIB_EXT)
+
+LIBSSH2_OPTS := $(CMAKE_COMMON) -DBUILD_SHARED_LIBS=ON -DBUILD_EXAMPLES=OFF \
+				-DCMAKE_PREFIX_PATH=$(build_prefix) \
+				-DCMAKE_INSTALL_RPATH=$(build_prefix) -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE \
+				-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LIBDIR=lib
+
+ifeq ($(OS),WINNT)
+LIBSSH2_OPTS += -DCRYPTO_BACKEND=WinCNG -DENABLE_ZLIB_COMPRESSION=OFF
+ifeq ($(BUILD_OS),WINNT)
+LIBSSH2_OPTS += -G"MSYS Makefiles"
+endif
+else
+LIBSSH2_OPTS += -DCRYPTO_BACKEND=mbedTLS -DENABLE_ZLIB_COMPRESSION=ON
+endif
+
+$(BUILDDIR)/$(LIBSSH2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR)/CMakeLists.txt $(MBEDTLS_OBJ_TARGET)
+	-cd $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/libssh2-mbedtls.patch
+	mkdir -p $(dir $@)
+	cd $(dir $@) && \
+	$(CMAKE) $(dir $<) $(LIBSSH2_OPTS)
+	touch -c $@
+
+$(LIBSSH2_OBJ_SOURCE): $(BUILDDIR)/$(LIBSSH2_SRC_DIR)/Makefile
+	$(MAKE) -C $(dir $<) libssh2
+	touch -c $@
+
+$(BUILDDIR)/$(LIBSSH2_SRC_DIR)/checked: $(LIBSSH2_OBJ_SOURCE)
+ifeq ($(OS),$(BUILD_OS))
+	$(MAKE) -C $(dir $@) test
+endif
+	echo 1 > $@
+
+$(LIBSSH2_OBJ_TARGET): $(LIBSSH2_OBJ_SOURCE) | $(build_shlibdir)
+ifeq ($(OS),WINNT)
+	$(MAKE) -C $(BUILDDIR)/$(LIBSSH2_SRC_DIR) install
+else
+	$(call make-install,$(LIBSSH2_SRC_DIR),)
```

Comment:
Ok. Thanks for being up late on a friday working through this. I'll try using the staged `make-install` and see if it works in cygwin for me, if so I'll change it to just BUILD_OS under the assumption that it might have just been an MSYS2 issue.